core/src/test/scala/integration/kafka/server/FetchFromFollowerIntegrationTest.scala

```diff
@@ -249,6 +249,9 @@ class FetchFromFollowerIntegrationTest extends BaseFetchRequestTest {
         reassignments.put(new TopicPartition(topicWithSingleRackPartitions, p), util.Optional.of(newAssignment))
       }
       admin.alterPartitionReassignments(reassignments).all().get(30, TimeUnit.SECONDS)
+      TestUtils.waitUntilTrue(
+        () => admin.listPartitionReassignments().reassignments().get().isEmpty,
```

Comment:
Do we want to pass in a timeout to `get` in case it hangs?

```suggestion
        () => admin.listPartitionReassignments().reassignments().get(30, TimeUnit.SECONDS).isEmpty,
```